Code/Framework/AzCore/AzCore/Math/Geometry3DUtils.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/Geometry3DUtils.h>
+#include <AzCore/std/function/function_template.h>
+#include <AzCore/Casting/numeric_cast.h>
+
+namespace AZ::Geometry3DUtils
+{
+    AZStd::vector<AZ::Vector3> GenerateIcoSphere(const uint8_t subdivisionDepth)
+    {
+        // The algorithm for generating an icosphere is to start with a regular icosahedron (20-sided polygon with triangluar faces),
+        // and then for each triangle, subdivide it further into 4 triangles at the midpoint of each edge. This is repeated recursively
+        // to the given depth to reach a desired number of subdivided faces.
+
+        constexpr int IcosahedronFaces = 20;
+
+        // "phi" is also known as the "golden ratio", (1 + sqrt(5)) / 2
+        const float phi = (1.0f + sqrt(5.0f)) * 0.5f;
```

Comment:
nit: you likely want `AZStd::sqrt` here to get the correct overloaded version, not the `double` version from the C standard library

Reply:
Fixed!